R/mlPrediction.R

```diff
@@ -315,7 +255,7 @@ is.jaspMachineLearning <- function(x) {
     return(jaspResults[["predictions"]]$object)
   } else {
     if (ready) {
-      colnames(dataset) <- decodeColNames(colnames(dataset))
+      dataset <- dataset[, which(colnames(dataset) %in% model[["jaspVars"]][["encoded"]]$predictors)]
```

Comment:
```suggestion
      dataset <- dataset[which(colnames(dataset) %in% model[["jaspVars"]][["encoded"]]$predictors)]
```
because if there is only one predictor
```r
df <- data.frame(x = 1:5, y = 6:10)
df[, c("x")] # vector
[1] 1 2 3 4 5
> df[c("x")] # data.frame
  x
1 1
2 2
3 3
4 4
5 5
```